renderer/hocs/withNumberValidation.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import * as yup from 'yup'
+import getDisplayName from '@zap/utils/getDisplayName'
+
+const isEmptyAmount = amount => amount === undefined || amount === null || amount === ''
+
+/**
+ * withNumberValidation - A HOC that will add validation of a `required` property to a field.
+ *
+ * @param {React.Component} Component Component to wrap
+ * @returns {React.Component} Wrapped component
+ */
+const withNumberValidation = Component =>
+  class extends React.Component {
```

Comment:
ideally, i'd use functional components for new components (since classes are deprecated sorts of)

Reply:
Good call. I've converted this.